src/java/org/apache/ivy/plugins/signer/bouncycastle/OpenPGPSignatureGenerator.java

```diff
@@ -161,7 +159,7 @@ private PGPSecretKey readSecretKey(InputStream in) throws IOException, PGPExcept
                     key = k;
                 }
                 if ((keyId != null)
-                        && (Long.valueOf(keyId, 16).longValue() == (k.getKeyID() & MASK))) {
+                        && (Long.valueOf(keyId, 16) == (k.getKeyID() & MASK))) {
```

Comment:
I haven't yet found an answer to this in some of the docs I quickly checked. I plan to check the Java specification later tonight. But if you can revert this one line from the rest of the PR, then I will go ahead and merge it since the rest of the PR is fine. I can read up about this a bit more without you having to wait for this PR to be merged.